compiler/hash-ast/src/parse.rs

```diff
@@ -77,31 +80,47 @@ enum EntryPoint<'a> {
 }
 
 pub struct ParParser<B> {
+    /// Number of workers the parser should use to parse a job.
     worker_count: NonZeroUsize,
+    /// Whether or not to visualise the generated AST
+    visualise: bool,
+    /// What backend to use for parsing, whether PEST or self hosted.
     backend: B,
 }
 
 impl<'c, B> ParParser<B>
 where
     B: ParserBackend<'c>,
 {
-    pub fn new(backend: B) -> Self {
-        Self::new_with_workers(backend, NonZeroUsize::new(num_cpus::get()).unwrap())
+    pub fn new(backend: B, visualise: bool) -> Self {
+        Self::new_with_workers(
+            backend,
+            NonZeroUsize::new(num_cpus::get()).unwrap(),
+            visualise,
+        )
     }
 
-    pub fn new_with_workers(backend: B, worker_count: NonZeroUsize) -> Self {
+    pub fn new_with_workers(backend: B, worker_count: NonZeroUsize, visualise: bool) -> Self {
         Self {
             worker_count,
             backend,
+            visualise,
         }
     }
 
+    pub fn set_visualisation(&mut self, visualise: bool) {
+        self.visualise = visualise;
+    }
```

Comment:
Is this needed? Why would one want to change the visualisation of an already constructed parser?